be/src/runtime/http_result_writer.cpp

```diff
@@ -0,0 +1,173 @@
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "runtime/http_result_writer.h"
+
+#include <column/column_helper.h>
+
+#include "column/chunk.h"
+#include "column/const_column.h"
+#include "exprs/cast_expr.h"
+#include "exprs/expr.h"
+#include "rapidjson/stringbuffer.h"
+#include "rapidjson/writer.h"
+#include "runtime/buffer_control_block.h"
+#include "runtime/current_thread.h"
+#include "types/logical_type.h"
+
+namespace starrocks {
+
+HttpResultWriter::HttpResultWriter(BufferControlBlock* sinker, const std::vector<ExprContext*>& output_expr_ctxs,
+                                   RuntimeProfile* parent_profile)
+        : _sinker(sinker), _output_expr_ctxs(output_expr_ctxs), _parent_profile(parent_profile) {}
+
+Status HttpResultWriter::init(RuntimeState* state) {
+    _init_profile();
+    if (nullptr == _sinker) {
+        return Status::InternalError("sinker is NULL pointer.");
+    }
+
+    return Status::OK();
+}
+
+void HttpResultWriter::_init_profile() {
+    _append_chunk_timer = ADD_TIMER(_parent_profile, "AppendChunkTime");
+    _convert_tuple_timer = ADD_CHILD_TIMER(_parent_profile, "TupleConvertTime", "AppendChunkTime");
+    _result_send_timer = ADD_CHILD_TIMER(_parent_profile, "ResultRendTime", "AppendChunkTime");
+    _sent_rows_counter = ADD_COUNTER(_parent_profile, "NumSentRows", TUnit::UNIT);
+}
+
+Status HttpResultWriter::append_chunk(Chunk* chunk) {
+    return Status::NotSupported("HttpResultWriter doesn't support non-pipeline engine");
+}
+
+Status HttpResultWriter::close() {
+    COUNTER_SET(_sent_rows_counter, _written_rows);
+    return Status::OK();
+}
+
+StatusOr<TFetchDataResultPtrs> HttpResultWriter::process_chunk(Chunk* chunk) {
+    SCOPED_TIMER(_append_chunk_timer);
+    int num_rows = chunk->num_rows();
+    std::vector<TFetchDataResultPtr> results;
+
+    Columns result_columns;
+    // Step 1: compute expr
+    int num_columns = _output_expr_ctxs.size();
+    result_columns.reserve(num_columns);
+
+    for (int i = 0; i < num_columns; ++i) {
+        ASSIGN_OR_RETURN(ColumnPtr column, _output_expr_ctxs[i]->evaluate(chunk));
+        column = _output_expr_ctxs[i]->root()->type().type == TYPE_TIME
+                         ? ColumnHelper::convert_time_column_from_double_to_str(column)
```

Comment:
what's the speciality of `time`?

Reply:
this is same behaivor with mysql protocal. TYPE_TIME's column is double, we have to translate double to string for readability